### PR TITLE
[7052] Added validation for course age range values

### PR DESCRIPTION
**Pull request by defong**
### Context
Validation for course age range

### Changes proposed in this pull request
Added validation for course age range

### Guidance to review
There is validation for course age range

### CODE REVIEW

**Comment by defong** [comment]
[Validate `course_age_range` values](https://trello.com/c/6o0kwkwl/7052-validate-courseagerange-values)

File: app/models/api/hesa_trainee_detail_attributes/v01.rb [hunk 1]
```diff
@@ -36,6 +36,8 @@ class V01
       end
 
       validates(*REQUIRED_ATTRIBUTES, presence: true)
+
+      validates(:course_age_range, inclusion: Hesa::CodeSets::AgeRanges::MAPPING.keys, allow_blank: true)
```
[the remark this answers is not in the document]
Comment (defong) [reply app/models/api/hesa_trainee_detail_attributes/v01.rb:40]: Without a model it kinda really going against `rails`
:shrug:

File: spec/requests/api/v0.1/post_hesa_trainees_spec.rb [hunk 2]
```diff
@@ -207,21 +208,41 @@
...
     end
+
+    context "course_age_range is empty" do
+      let(:params) { { data: } }
+
+      let(:course_age_range) { "" }
+
+      it "return status code 422 with a meaningful error message" do
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.parsed_body["errors"]).to contain_exactly("Course age range can't be blank")
```
[the remark this answers is not in the document]
Comment (tomtrentham) [reply spec/requests/api/v0.1/post_hesa_trainees_spec.rb:232]: I think the requirement for this field to be present comes from Product/BA. It's always been required in the spreadsheets that we've worked from, and the API docs indicate that too.

[the remark this answers is not in the document]
Comment (tomtrentham) [reply spec/requests/api/v0.1/post_hesa_trainees_spec.rb:232]: Interesting. The HESA data that the API is currently asking for seems to be a [required HESA field](https://www.hesa.ac.uk/collection/c23053/e/ittphsc) so I think we'll be ok for now, but might need to change this once the API is opened up to non-HEIs.

**Review by tomtrentham — APPROVED** [review]

**Pull request merged**